README.md

```diff
@@ -49,7 +49,245 @@ kubebuilder docs
+
+```sh
+kubebuilder create resource --group mygroup --version v1beta1 --kind MyKind
+```
+
+### pkg/controllers
+
+**Users must packages under this package**
+
+The `controllers` package contains the controllers to implement the resource APIs.  Controllers are defined under
+`pkg/controllers/<kind>/controller.go`.  Controllers may be annotated with comments to wire the controller into the
+inject package, start informers they require and install RBAC rules they require.
+
+Subpackages of controllers are created when running the following command to create a new resource and controller:
+
+```sh
+kubebuilder create resource --group mygroup --version v1beta1 --kind MyKind
+```
+
+See documentation and examples of annotations in godoc format here:
+- [gen/controller](https://godoc.org/github.com/kubernetes-sigs/kubebuilder/pkg/gen/controller#example-package)
+
+### pkg/inject
+
+*Most users do not need to edit this package.*
+
+The `inject` package contains the `RunAll` function used to start all of the registered controllers and informers.
+Wiring is autogenerated based on resource and controller annotations.
+
+Generated wiring:
+- Installing CRDs
+- Instantiating and starting controllers
+- Starting sharedinformers
+- Installing RBAC rules
+
+### pkg/inject/args
+
+*Only some users need to edit this package.*
+
+The `args` package contains the struct passed to the `ProvideController` function used to instantiating controllers.
+The `InjectArgs` struct and `CreateInjectArgs` function in this package may be edited to pass additional information
+to the controller provider functions.  This is typically an advanced use case.
+
+The `args` package uses the `kubebuilder/pkg/inject/args` package documented here:
+[inject/args](https://godoc.org/github.com/kubernetes-sigs/kubebuilder/pkg/inject/args)
+
+### hack/
+
+The `hack` directory contains generated samples and config.
+
+The hack/install.yaml config file for installing the API extension into a cluster will be created under this
+directory by running.
+
+```sh
+kubebuilder create config --controller-image mycontrollerimage --name myextensionname
+```
+
+### docs/
+
+The `docs` package contains your examples and content for generating reference documentation for your APIs.
+
+The docs package is created when running either
+
+```sh
+kubebuilder docs
+```
+
+or
+
+```sh
+kubebuilder create example --group mygroup --version v1beta1 --kind MyKind
+```
+
+Example reference documentation lives under `docs/reference/examples`.  Conceptual reference documentation
+lives under `docs/reference/static_includes`.
+
+### /
+
+The project root directory contains several files
+
+- Dockerfile.controller
+
+Running `docker build` on this file will build a container image for running your controller.
+
+- Gopkg.toml / Gopkg.lock
+
+These files are used to update vendored go dependencies.
+
+## Common controller operations
```

Comment:
Maybe `Common controller operations available`?  Since following paragraphs explained the controller operations and the corresponding library function.